app/js/app/directives/react_classes/RegexPatternBlock.js

```diff
@@ -0,0 +1,78 @@
+define(["react", "jquery"], function(React,$) {
+	return function(_ContentEditor, Block, ContentBlock, ContentValueOrChildren) {
+		return React.createClass({
+
+			onDocChange: function(_c, oldDoc, newDoc) {
+				this.props.onChange(this, oldDoc, newDoc);
+			},
+
+			onValueChange: function(e) {
+				var oldDoc = this.props.doc;
+				var newDoc = $.extend({}, oldDoc);
+				newDoc.value = e.target.value;
+				this.onDocChange(this, oldDoc, newDoc);
+			},
+
+			onExplanationChange: function(_c, _oldVal, newVal) {
+				var oldDoc = this.props.doc;
+				var newDoc = $.extend({}, oldDoc);
+				newDoc.explanation = newVal;
+				this.onDocChange(this, oldDoc, newDoc);
+			},
+
+			caseInsensitive_toggle: function(_e) {
+				var oldDoc = this.props.doc;
+				var newDoc = $.extend({}, oldDoc);
+				newDoc.caseInsensitive = !oldDoc.caseInsensitive;
+				this.onDocChange(this, oldDoc, newDoc);
+			},
+
+			multiLineRegex_toggle: function(_e) {
+				var oldDoc = this.props.doc;
+				var newDoc = $.extend({}, oldDoc);
+				newDoc.multiLineRegex = !oldDoc.multiLineRegex;
+				this.onDocChange(this, oldDoc, newDoc);
+			},
+
+			correct_toggle: function(_e) {
+				var oldDoc = this.props.doc;
+				var newDoc = $.extend({}, oldDoc);
+				newDoc.correct = !oldDoc.correct;
+				this.onDocChange(this, oldDoc, newDoc);
+			},
+
+			render: function() {
+				var emptyExplanation = {
+					type: "content",
+					children: [],
+					encoding: "markdown"
+				};
+				return (
+					<Block type="content" blockTypeTitle={this.props.blockTypeTitle} doc={this.props.doc} onChange={this.onDocChange}>
+						<div className="row">
+							<div className="small-1 column text-right">
+								{this.props.doc.correct ?
+									<i style={{color: "#0a0"}} className="correct-mark general foundicon-checkmark" onClick={this.correct_toggle}/> :
+									<i style={{color: "#a00"}} className="correct-mark general foundicon-remove" onClick={this.correct_toggle} />}
+							</div>
+							<div className="small-6 columns" >
+								<input type="text" value={this.props.doc.value || ""} onChange={this.onValueChange} />
+								<div className="row">
```

Comment:
The way I've separated the checkboxes with bootstrap is a bit verbose - if there's a shorter way please tell me!

Reply:
This is how we generally separate them out, but instead of a function for each of the checkbox_toggles, you could perhaps mirror the `onCheckboxChange` function (line 126 of `QuestionBlock.js`)

Reply:
Good point, I've made the change and code neatness has improved by about 31.42%